rotkehlchen/chain/evm/decoding/balancer/utils.py

```diff
@@ -0,0 +1,223 @@
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_balancer_v1_data(
+        inquirer: 'EvmNodeInquirer',
+        cache_type: Literal[CacheType.BALANCER_V1_POOLS],
+) -> None:
+    """Query and store balancer v1 pools.
+    May raise:
+    - RemoteError
+    """
+    latest_pools_count = _query_balancer_pools_count(
+        chain=inquirer.chain_id,
+        version=1,
+    )
+    cache_key_parts = (cache_type, inquirer.chain_name, '1')
+    now = ts_now()
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        if (saved_pools_count_str := globaldb_get_unique_cache_value(
+            cursor=cursor,
+            key_parts=cache_key_parts,
+        )) is None:
+            saved_pools_count = 0
+        else:
+            saved_pools_count = int(saved_pools_count_str)
+
+    if latest_pools_count == saved_pools_count:
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value_at_ts(
+                write_cursor=write_cursor,
+                key_parts=cache_key_parts,
+                value=str(latest_pools_count),
+                timestamp=now,
+            )
+            return None
+
+    pools_data = _query_balancer_pools(
+        chain=inquirer.chain_id,
+        database=inquirer.database,
+        version=1,
+    )
+    for pool in pools_data:
+        try:
+            _create_or_update_pool_token(
+                db=inquirer.database,
+                chain_id=inquirer.chain_id,
+                protocol=CPT_BALANCER_V1,
+                pool=pool,
+            )
+        except (KeyError, ValueError, TypeError) as e:
+            log.error(
+                f'Failed to process balancer v1 pool due to {e!s}. '
+                f'Pool: {pool}. Skipping...',
+            )
+            continue
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_unique_cache_value_at_ts(
+            write_cursor=write_cursor,
+            key_parts=cache_key_parts,
+            value=str(latest_pools_count),
+            timestamp=now,
+        )
+
+
+def _query_balancer_pools_count(chain: 'ChainID', version: Literal[1, 2]) -> int:
+    """Fetch the total number of pools for the specified chain and protoco
+    May raise:
+        - RemoteError
+        - DeserializationError
+    """
+    try:
+        response = requests.post(
+            url=BALANCER_API_URL,
+            json={
+                'variables': {
+                    'chain': CHAIN_ID_TO_BALANCER_API_MAPPINGS[chain],
+                    'version': version,
+                },
+                'query': GET_POOLS_COUNT_QUERY,
+            },
+            timeout=CachedSettings().get_timeout_tuple(),
+        )
+        response.raise_for_status()
+        data = response.json()
+    except (requests.exceptions.RequestException, JSONDecodeError) as e:
+        raise RemoteError(f'Balancer pools count query failed due to {e!s}') from e
```

Comment:
can you check what I'm going to say but:

- JSONDecodeError is different from requests.exception.JSONDecodeError
- you can use requests.RequestException
- requests.RequestException is a parent class of requests.JSONDEcoderError

Reply:
okay, thanks